### ISSUE TITLE
[3.6.0] Nested arrays no longer returned from nested AQL queries

### ISSUE THREAD

**Issue by ysmu** [arangodb/arangodb#10852]
## My Environment

* __ArangoDB Version__:        3.6.0
* __Storage Engine__:             RocksDB
* __Deployment Mode__:       Single Server
* __Deployment Strategy__:   Manual Start in Docker
* __Configuration__:               Nothing special
* __Infrastructure__:               own
* __Operating System__:        Windows 10
* __Total RAM in your machine__:        16Gb
* __Disks in use__:        SSD
* __Used Package__:              Docker - official Docker library

## Steps to reproduce

This looks like a regression from 3.5 (probably due to the new subquery splicing optimization)

AQL
```js
LET results = (
    FOR doc IN []
        LET docs = (
            FOR doc2 in []
                RETURN doc2
        )

        RETURN [docs]
)

RETURN [results]
```

Result (3.5.2) - Expected result
```js
[
  [
    []
  ]
]
```

Result (3.6.0) - Bug?
```js
[]
```

Profile (3.5.2)
```
Execution plan:
 Id   NodeType            Calls   Items   Runtime [s]   Comment
  1   SingletonNode           1       1       0.00000   * ROOT
 12   SubqueryNode            1       1       0.00005     - LET results = ...   /* const subquery */
  2   SingletonNode           1       1       0.00000       * ROOT
  3   CalculationNode         1       1       0.00000         - LET #8 = [ ]   /* json expression */   /* const assignment */
  4   EnumerateListNode       1       0       0.00004         - FOR doc IN #8   /* list iteration */
  9   SubqueryNode            1       0       0.00000           - LET docs = ...   /* const subquery */
  5   SingletonNode           0       0       0.00000             * ROOT
  6   CalculationNode         0       0       0.00000               - LET #10 = [ ]   /* json expression */   /* const assignment */
  7   EnumerateListNode       0       0       0.00000               - FOR doc2 IN #10   /* list iteration */
  8   ReturnNode              0       0       0.00000                 - RETURN doc2
 10   CalculationNode         1       0       0.00000           - LET #12 = [ docs ]   /* simple expression */
 11   ReturnNode              1       0       0.00000           - RETURN #12
 13   CalculationNode         1       1       0.00000     - LET #14 = [ results ]   /* simple expression */
 14   ReturnNode              1       1       0.00000     - RETURN #14

Indexes used:
 none

Optimization rules applied:
 none
```

Profile (3.6.0)
```
Execution plan:
 Id   NodeType            Calls   Items   Runtime [s]   Comment
  1   SingletonNode           1       1       0.00000   * ROOT
 15   SubqueryStartNode       1       2       0.00000     - LET results = ( /* subquery begin */
  3   CalculationNode         1       2       0.00000       - LET #8 = [ ]   /* json expression */   /* const assignment */
  4   EnumerateListNode       1       1       0.00001       - FOR doc IN #8   /* list iteration */
 17   SubqueryStartNode       1       1       0.00001         - LET docs = ( /* subquery begin */
  6   CalculationNode         1       1       0.00000           - LET #10 = [ ]   /* json expression */   /* const assignment */
  7   EnumerateListNode       1       1       0.00002           - FOR doc2 IN #10   /* list iteration */
 18   SubqueryEndNode         2       1       0.00004         - RETURN  doc2 ) /* subquery end */
 10   CalculationNode         2       1       0.00000         - LET #12 = [ docs ]   /* simple expression */
 16   SubqueryEndNode         1       0       0.00004     - RETURN  #12 ) /* subquery end */
 13   CalculationNode         1       0       0.00000     - LET #14 = [ results ]   /* simple expression */
 14   ReturnNode              1       0       0.00000     - RETURN #14

Indexes used:
 none

Optimization rules applied:
 Id   RuleName
  1   splice-subqueries
```

**Comment by ysmu** [comment]
The issue goes away in 3.6.0 once I disable the `splice-subqueries` optimization rule.

**Comment by goedderz** [comment]
Reproduced.

**Comment by adeoy** [comment]
I have a similar issue when use `UPSERT` inside a `LET` clause expecting return a `NEW._id`.
```javascript
LET new_id = (
    UPSERT { _key: @_key }
    INSERT { field_1: @field_1, field_2: @field_2  }
    UPDATE { field_1: @field_1, field_2: @field_2  }
    IN my_collection
    RETURN NEW._id
)
RETURN new_id
```

Expected:
```
[
  "my_collection/12345678"
]
```

Returned:
```
[]
```

I dont know how to disable this `splice-subqueries`.

**Comment by Simran-B** [comment]
@adeoy You can disable it by default via startup option:

    --query.optimizer-rules "-splice-subqueries"

Or via query option, e.g. in arangosh:

    db._query(`<query>`, {}, { optimizer: { rules: ["-splice-subqueries"] } })

**Comment by goedderz** [comment]
@ysmu thanks for your precise bug report. This will be fixed in 3.6.1.

And thanks to you too, @adeoy, this was helpful and will also be fixed in 3.6.1.

**Comment by adeoy** [comment]
> @adeoy You can disable it by default via startup option:
> 
> ```
> --query.optimizer-rules "-splice-subqueries"
> ```
> 
> Or via query option, e.g. in arangosh:
> 
> ```
> db._query(`<query>`, {}, { optimizer: { rules: ["-splice-subqueries"] } })
> ```

Thanks! I with this tweak the 3.6.0 version works fine.
